File: windows/MicroKeys.Core/Localization.cs
namespace MicroKeys.Core;

/// <summary>The two UI languages MicroKeys ships with.</summary>
public enum Language { ZhHans, En }

/// <summary>
/// Process-wide current language. The app sets it from its saved preference;
/// Core's error messages read it so config errors come out in the right language.
/// </summary>
public static class L10n
{
    public static Language Language { get; set; } = SystemDefault;

    /// <summary>Chinese if the user's UI culture is any Chinese variant, English otherwise.</summary>
    public static Language SystemDefault =>
        System.Globalization.CultureInfo.CurrentUICulture.Name.StartsWith("zh", StringComparison.OrdinalIgnoreCase)
            ? Language.ZhHans : Language.En;

    public static string Pick(string zh, string en) => Language == Language.ZhHans ? zh : en;
}

File: macos/Sources/MicroKeysCore/Localization.swift
import Foundation

/// The two UI languages MicroKeys ships with.
public enum Language: String, CaseIterable {
    case zhHans = "zh-Hans"
    case en = "en"

    /// What the system prefers: Chinese if the user's first preferred language
    /// is any Chinese variant, English otherwise.
    public static var systemDefault: Language {
        let first = Locale.preferredLanguages.first?.lowercased() ?? "en"
        return first.hasPrefix("zh") ? .zhHans : .en
    }
}

/// Process-wide current language. The app sets it from its saved preference;
/// Core's error messages read it so config errors come out in the right language.
public enum L10n {
    public static var language: Language = .systemDefault

    public static func pick(_ zh: String, _ en: String) -> String {
        language == .zhHans ? zh : en
    }
}
